enhancements/network/service-egress-traffic-steering.md

```diff
@@ -159,38 +169,50 @@ the egress traffic flow for the pod `10.244.1.6` on `ovn-worker2` towards an ext
-When an `EgressService` resource is created for a LoadBalancer service, a node is selected to be in charge of all of its traffic.
-If the `nodeSelector` field is specified, only a node whose labels match the specified selectors can be selected.
+When an `EgressService` resource is created for a LoadBalancer service with a valid `snat` field, a node is selected to be in charge of all of its traffic.
+If the `snat.nodeSelector` field is specified, only a node whose labels match the specified selectors can be selected.
 A cache of the nodes and their number of allocations is kept in order to spread the allocations between all of the nodes available for a given service - selecting the node with the least amount of allocations every time.
 
 Once a node is selected, it is checked for readiness to serve traffic every x seconds the same way EgressIP does for its nodes (TCP/gRPC).
 If a node fails the health check, its allocations move to another node by removing the `egress-service.k8s.ovn.org/<svc-namespace>-<svc-name>: ""` label from it, removing the logical router policies from the cluster router and resetting the status of the `EgressService` which causes it to be reconciled - causing a new node to be selected for the service.
 If the node becomes not ready or its labels no longer match the service's selectors the same re-election process happens.
 
+#### Network without SNAT
+As mentioned earlier, it is possible to use the "Network" capability without SNATing the traffic to the service's ingress IP. This is done by creating an EgressService with the `Network` field specified **without** specifying the `snat` field.
+
+An EgressService without SNAT does not need to have a host selected, as the traffic will exit each node with the IP of the interface corresponding to the "Network" by leveraging the masquerade rules that are already in place.
+
+This works only on clusters running on "Local" gateway mode, because on "Shared" gateway mode the ip rules created by the controller are ignored (like all of the node's routing stack).
+
+When the `snat` field is not specified, `ovnkube-master` does not need to create any logical router policies as the egress packets of each pod would exit through the pod's node.
+However, `ovnkube-master` will set the status field of the resource with `host: ALL` to designate that no reroute logical router policies exist for the service, "instructing" all of the `ovnkube-nodes` to handle the resource's `Network` field without creating SNAT iptables rules.
+
+When `ovnkube-node` detects that the host of an EgressService is `ALL`, only the endpoints local to the node will have an ip rule created, and no SNAT iptables rules will be created.
+
+It is the user's responsibility to make sure that the pods backing an EgressService without SNAT run only on nodes that have the required "Network", as no additional steering (lrps) will take place by OVN and pods running on nodes without a correct "Network" will misbehave.
```

Comment:
Can you help me understand why do you want to get rid of creating the LRPs if `snat` is not specified?
This breaks `network` for shared gw mode and adds additional restrictions on where the pods can be scheduled.

Reply:
The main disadvantage of using snat is the host being a "bottleneck". Using networks it feels more natural (and better) to free it up and let every node handle its own traffic (as usual) instead of forcing a pretty major limitation that shouldn't really be present. In any case, since the network field relies on the node's routing stack I think it is fine to "break" it on sgw since users shouldn't be modifying networks on nodes anyways - that is, afaik the users interested in this feature are ones that already strictly use lgw and supporting it on sgw would require special care that is probably not necessary at this point.

Reply:
I am ok to leave it as is for now. Even though I consider not being able to specify a `nodeSelector` for `network` while it is possible to do for `snat` a limitation, the feature is still in tech preview and the current API allows us to add the support in the future. Additionally it allows us to simplify the implementation since there is no need to create custom OVN flows for this usecase.